bundles/org.openhab.binding.tado/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -24,6 +24,14 @@
 				<description>Password of tado login used for API access</description>
 				<context>password</context>
 			</parameter>
+
+			<parameter name="maxReconnectAttempts" type="integer" min="0" max="10" step="1" required="true">
```

Comment:
The home handler Bridge has a polling interval of 5 minutes so 10 re-connect attempts equals 50 minutes. I suppose I could increase the maximum to (say) 12 attempts, equals 60 minutes, but I don't see any point in extending it any longer than that.

After the last failed reconnect attempt the thing remains permanently offline, which IMHO gives a permanent indication in the UI that there is probably something physically wrong with that device (e.g. powered off).

Furthermore if the tado cloud server goes down, the home handler Bridge goes offline, which will prevent it from doing any reconnection attempts until the cloud server comes back.

Reply:
I was thinking to have some value indicating "indefinitely". When the device is powered on again, it will be reconnected without any manual interaction. And it only costs one attempt every 5 minutes? I might be missing something, but what is the argument for giving up?

Reply:
I made the following changes..
- increased maxReconnectAttempts from 10 to 12
- added a server ping in the bridge polling code to ensure that bridge goes offline if server is down

Reply:
My prior post crossed with yours.

> I was thinking to have some value indicating "indefinitely"

Yes, that makes sense. Perhaps something like this?
- 0 = never try to reconnect
- 1..12 = maximum number of reconnection attempts
- -1 = unlimited reconnection attempts

Reply:
Done

Reply:
Sorry, one more comment. :-) Is there any disadvantage in trying indefinitely?

If not, from a user perspective this should be the default, and I would also suggest to hide the option as advanced.

Also, to simplify, perhaps 0 could be "unlimited" instead of -1, and there would simply be no possibility to avoid retries, but advanced users could still limit it to 1.

It might also be a possibility to simply always retry indefinitely without any configuration parameter. This would simplify further, but of course you can leave the parameter if it has some value/advantage for specific scenarios and/or for advanced users.

Reply:
> Is there any disadvantage in trying indefinitely?

I guess not. In that case I will remove the config param, and respective changes in things.xml, i18n, ReadMe, etc. — It makes the PR very simple indeed. :)

Reply:
I have reverted to making indefinite reconnect attempts; this reduces the scope of changes from six files down to just one. :)